doc/ref/chap-osh-assign.md

```diff
@@ -73,20 +73,19 @@ applied.
 In the first phase, the type adjustment is performed in the following way:
 
 - When the LHS variable is unset, the assignment creates an empty indexed array
```

Comment:
We can talk about it later, but it is perhaps confusing to say indexed array (BashArray) -- i.e. to have two different terms

For now it's OK

Reply:
Which one should we prefer in the documentation?

Reply:
I think we should try to make all of these match

- the type names (user visible types - it would be nice if internal types match too though)
- the help topic names
- the terms used in the text

We can talk about it on Zulip -- but it is not essential right now

It is probably more important to see if this can run any new tests in ble.sh and so forth, and then we can circle back to the names if necessary

The names now are decent, but I think could be improved a bit.

At least I may want to add a `bash-` or `sh-` prefix consistently , with `sh-` meaning "POSIX shell"  ...  Although some things like `+=` are not limited to bash